### ISSUE TITLE
Unexpected behaviour from at least isEqual() when using UNIX epoch date (1970-01-01)

### ISSUE THREAD

**Issue by wodow** [spencermountain/spacetime#417]
```js
> spacetime('1970-01-01').isEqual('1970-01-01')
null
> spacetime('1970-01-02').isEqual('1970-01-02')
true
> spacetime('1970-01-02').isEqual('1970-01-01')
null
> spacetime('1970-01-01').isEqual('1970-01-02')
false
```

Should only return null when passed an invalid date?

**Comment by wodow** [comment]
Possible this affects other functions too, but have not yet checked.

**Comment by spencermountain** [comment]
hey Chris, thanks for the issue. I don't see any nulls, when I do this:
```js
console.log(spacetime('1970-01-01').isEqual('1970-01-01'))
console.log(spacetime('1970-01-02').isEqual('1970-01-02'))
console.log(spacetime('1970-01-02').isEqual('1970-01-01'))
console.log(spacetime('1970-01-01').isEqual('1970-01-02'))
```
same thing in the repl - how did you produce this null response? You're right, that it should be a boolean.
cheers

**Comment by spencermountain** [comment]
weird - i'm seeing true/false in chrome/node/firefox
![2024-10-09 T 13 35 02@2x](https://github.com/user-attachments/assets/9a39d03a-b02d-4bbb-988a-893e95046724)

**Comment by wodow** [comment]
I've worked out why: we are both avoiding the obvious and also super pertinent elephant in the room, the timezone!

It only behaves this way with a GMT/UTC/London timezone, where the epoch is equal to zero.

`getEpoch` should allow for `tmp.epoch` being zero, but it only checks for truthiness:

https://github.com/spencermountain/spacetime/blob/0464ecaebb935bc89fe2507fc4e0b478362c4a57/src/fns.js#L59-L71

**Comment by spencermountain** [comment]
haha, thank you. busted by pertinent elephant again.
Good catch. Happy to do a hotfix today, if I can.
cheers

**Comment by spencermountain** [comment]
released as `7.6.2`
thanks for your help!
